### ISSUE TITLE
Please whip up some documentation on how you process Yaru source for this

### ISSUE THREAD

**Issue by ahvth** [Jannomag/Yaru-Colors#88]
I'd like to put in a bit of work to solve #75 , but I would probably need to rebase the project on another version of Yaru.

@Jannomag I'm a bit curious as to what your method is for rebasing the project on a new release of the upstream, and I'd be happy to help update the src documentation so that others can also have a go at fixing issues when you don't have time.

Please excuse me if there are already any such instructions that I've overlooked.

Thanks!

**Comment by Jannomag** [comment]
Well, I tried my best to add notes to my theme-script.sh file.
I just looked over the compiler files from Yarus source and used the commands without needing the whole compilation process of Yaru, so I used sassc and sass for the css compiling and optipng and Inkscape for svg rendering. 
If it’s not easy to understand, it’s my bad. 
As it seems I need to rework everything for 21.04, again. If so, I’ll try to add more documentation.

I created the theme-script.sh for make it easier for me to compile the themes, mostly. It wasn’t meant to make it easy to create own themes.

And since I’m not a programmer and have absolutely no clue how complying works, how GitHub works and how I can make things more compact like Yarus source, for example, I’ll need time for it.

also, I did most of the things at work, but currently my boss is flooding me with work for a few weeks, so I don’t have as much time as I’ve had for the current versions.

Long story short, you’re absolutely free to adapt and modify my files if you like. You can also make pull requests if you find something to update. 
For now I just can recommend you to try to understand my theme-script.sh.

**Comment by ahvth** [comment]
Sorry, responded in the other issue thread before reading this. :facepalm: Thanks for the reply. This whole deal is indeed very complex. Makes it easy to understand why the gnome devs don't want to support themeing, but I think the results here speak for themselves in terms of a great product.

I wonder if there's a way I can add the scripts you've made for altering colors to Yaru's master branch instead of working backwards off their point releases. This would probably make it easier to keep step with new Yaru releases (and their current master branch is looking pretty good).

I'm no programmer either, by the way, so if I don't understand your well-commented scripts, it's not necessarily your fault. :smile:

**Comment by Jannomag** [comment]
It’s not complicated to change the colors, you just need to change the colors in the _colors.scss and _ubuntu-colors.scss (or what they are really named, don’t have it in mind atm).
The complex part is to change the color values in every svg file for assets, icons and cursors, because some use gradients (icons for example).
My script is as long as it is just because it can compile everything.

Yaru uses make for this, if I remember it correctly. So the compiling steps are written in multiple files while my script is just one file.

Since I don’t know how to create something like original Yaru source files, I created my script which is just batch.

It should be possible to create a complete new script, which clones the Yaru repo and compiles the colors instead of using local files like I did...if I would know how to this, I would do it.
But I don’t want to put so much time on it, tbh. I have a small daughter and too many hobbies...mostly I spend the free time I have at home with gaming and sometimes some arduino and electronics.
I also don’t use Ubuntu as my daily driver anymore because of many reasons.
